### ISSUE TITLE
commit 53db31 of deflation branch deleted a bunch of stuff

### ISSUE THREAD

**Issue by kostrzewa** [etmc/tmLQCD#328]
unfortunately a large number of things were removed during the merge of the deflation branch, in particular commit 53db31bff1d377816238945b127fdfccee97c741 caused a number of solvers to disappear...

**Comment by kostrzewa** [comment]
I think https://github.com/kostrzewa/tmLQCD/tree/mixed_cg_merge_hmc can serve as a guide on how to restore the lost functionality...

**Comment by kostrzewa** [comment]
It's not clear in the github diff, but it becomes clear when using ```git show```

**Comment by urbach** [comment]
Hmm, that's bad. Do you already know what to do? 

On 24 May 2016 20:32:02 CEST, Bartosz Kostrzewa <notifications@github.com> wrote:
>It's not clear in the github diff, but it becomes clear when using
>```git show```
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/etmc/tmLQCD/issues/328#issuecomment-221355573

Carsten Urbach, www.carsten-urbach.eu

**Comment by kostrzewa** [comment]
> Hmm, that's bad. Do you already know what to do?

My plan was to first restore the solver logic in ```invert_doublet_eo.[c,h]``` and ```operator.c```, which is where  I first noticed it. Afterwards we will have to watch for further regressions and do further manual merges. I don't think that it would be possible, without a major investment of effort, to completely reconstruct what happened and how it affected everything. There were just too many changes in the deflation pull-request to make this practical...

**Comment by kostrzewa** [comment]
update: I meant ```invert_clover_eo.[c,h]```

**Comment by urbach** [comment]
> unfortunately a large number of things were removed during the merge of the deflation branch, in particular commit 53db31bff1d377816238945b127fdfccee97c741 caused a number of solvers to disappear...

okay, I see that the RGMIXEDCG is no longer available in
invert_clover_eo.c. What else?


> 
> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/etmc/tmLQCD/issues/328
--
Carsten Urbach
e-mail: curbach@gmx.de
        urbach@hiskp.uni-bonn.de
Fon   : +49 (0)228 73 2379
skype : carsten.urbach
URL: http://www.carsten-urbach.eu

**Comment by urbach** [comment]
at least the following in missing in `invert_clover_eo.c` in the logic

    else if(solver_flag == RGMIXEDCG){
     iter = rg_mixed_cg_her(Odd_new, g_spinor_field[DUM_DERI], solver_params, max_iter, precision, rel_prec,
                                            VOLUME/2, &Qsw_pm_psi, &Qsw_pm_psi_32);
       Qm(Odd_new, Odd_new);
     }

**Comment by kostrzewa** [comment]
Well, for now it's only #329, but we'll have to see how things develop. Maybe other files were similarly affected...

**Comment by urbach** [comment]
yes, I agree, we have to review carefully. I have a boring meeting starting
at 11, where I might find some time...

--
Carsten Urbach
e-mail: curbach@gmx.de
        urbach@hiskp.uni-bonn.de
Fon   : +49 (0)228 73 2379
skype : carsten.urbach
URL: http://www.carsten-urbach.eu
